File: src/xianyu_cli/utils/url.py
"""URL helpers for Goofish item links."""

from __future__ import annotations

GOOFISH_ITEM_URL = "https://www.goofish.com/item?id={item_id}"


def item_url(item_id: str) -> str:
    """Build a Goofish item URL from an item ID."""
    return GOOFISH_ITEM_URL.format(item_id=item_id)

File: golang/utils/url.go
package utils

import "fmt"

const (
	goofishItemURL = "https://www.goofish.com/item?id=%s"
	goofishH5URL   = "https://h5.m.goofish.com/item?id=%s"
)

// ItemURL builds a Goofish item URL from an item ID.
func ItemURL(itemID string) string {
	return fmt.Sprintf(goofishItemURL, itemID)
}

// ShareURL builds a mobile-friendly Goofish share URL.
func ShareURL(itemID string) string {
	return fmt.Sprintf(goofishH5URL, itemID)
}
